manifests/load.pp

```diff
@@ -27,7 +36,16 @@
     }
 
     'absent': {
-      $changes = "rm '${name}'"
+      case $::osfamily {
+        'Debian': {
+          $changes = "rm '${name}'"
+        }
+        'Suse': {
+          $changes = "rm MODULES_LOADED_ON_BOOT/value[.='${name}']"
+          $onlyif  = "match MODULES_LOADED_ON_BOOT/value[.='${name}'] not_include ${name}"
+        }
+        default: { }
```

Comment:
Am 7/13/2015 um 11:16 AM schrieb Mickaël Canévet:
>
> In manifests/load.pp 
> <https://github.com/camptocamp/puppet-kmod/pull/35#discussion_r34445468>:
>
> > @@ -27,7 +36,16 @@
> >      }
> >
> >      'absent': {
> > -      $changes = "rm '${name}'"
> > +      case $::osfamily {
> > +        'Debian': {
> > +          $changes = "rm '${name}'"
> > +        }
> > +        'Suse': {
> > +          $changes = "rm MODULES_LOADED_ON_BOOT/value[.='${name}']"
> > +          $onlyif  = "match MODULES_LOADED_ON_BOOT/value[.='${name}'] not_include ${name}"
> > +        }
> > +        default: { }
>
> @buzzdeee <https://github.com/buzzdeee> why did you deliberately not 
> added here?
>

The $changes is not used by RedHat, therefore, I kept the default empty 
here, instead of failing.
Otherwise, I could add an empty RedHat statement, and fail in the 
default case.

Sebastian

> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/camptocamp/puppet-kmod/pull/35/files#r34445468>.
>